### PR TITLE
Source GoogleAds: add end_date to config

### PR DESCRIPTION
**Pull request by iberchid**
## What
Relates to issue #8666 

## How
Added end_date to spec and updated the source.py and streams.py files to allow for setting an end_date.

## Recommended reading order
1. `spec.json`
2. `source.py`
3. `streams.py`

## 🚨 User Impact 🚨
No changes, because when the parameter is not provided the default value is used which is identical to the previous implementation.

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 
   

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [x] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the new connector version is published, connector version bumped in the seed directory as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</p>
</details>

<details><summary> <strong> Connector Generator </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

---
This change is [Reviewable](https://reviewable.io/reviews/airbytehq/airbyte/8669)

### CODE REVIEW

File: airbyte-integrations/connectors/source-google-ads/source_google_ads/source.py [hunk 1]
```diff
@@ -56,8 +56,9 @@ def check_connection(self, logger: AirbyteLogger, config: Mapping[str, Any]) ->
 
     def streams(self, config: Mapping[str, Any]) -> List[Stream]:
         google_api = GoogleAds(credentials=self.get_credentials(config), customer_id=config["customer_id"])
+        end_date = config["end_date"] if "end_date" in config else None
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-google-ads/source_google_ads/source.py:59]:
```suggestion
        end_date = config.get("end_date")
```

File: airbyte-integrations/connectors/source-google-ads/source_google_ads/streams.py [hunk 2]
```diff
@@ -64,19 +64,22 @@ class IncrementalGoogleAdsStream(GoogleAdsStream, ABC):
...
+    def __init__(self, start_date: str, conversion_window_days: int, end_date: str=None, **kwargs):
         self.conversion_window_days = conversion_window_days
         self._start_date = start_date
+        self._end_date = end_date
         super().__init__(**kwargs)
 
     def stream_slices(self, stream_state: Mapping[str, Any] = None, **kwargs) -> Iterable[Optional[Mapping[str, any]]]:
         stream_state = stream_state or {}
         start_date = stream_state.get(self.cursor_field) or self._start_date
+        end_date = stream_state.get(self.cursor_field) or self._end_date
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-google-ads/source_google_ads/streams.py:78]:
```suggestion
        end_date = self._end_date
```
if you have a cursor_field the start date and end date will become equal and you don't want to do this.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
See comments.

**Comment by marcosmarxm** [comment]
@iberchid please run unit tests to validate your change also please add a new unit test with a specific end_date to validate your changes.

**Comment by marcosmarxm** [comment]
@sherifnada can you review the UX modification this PR makes?

**Comment by iberchid** [comment]
Hi @marcosmarxm , Thanks for your comments, I updated the scripts according to your recommendations. I also ran existing unit tests and added a few more:

image

**Comment by marcosmarxm** [comment]
@iberchid can you solve the conflicts and make sure the `./gradlew airbyte-integrations:connectors:source-google-ads:integrationTest` is working?

**Comment by marcosmarxm** [comment]
@iberchid can you solve the conflicts?

**Comment by marcosmarxm** [comment]
You need to create a folder called `secrets` and add a file inside with the config to run integration tests and check connection.

**Comment by iberchid** [comment]
Here it is @marcosmarxm :

image

**Comment by iberchid** [comment]
Hi @marcosmarxm , what is the status of this PR? Are there any other TODOs left for me?

**Review by marcosmarxm — APPROVED** [review]

**Pull request merged**